org.eclipse.lsp4e/src/org/eclipse/lsp4e/progress/LSPProgressManager.java

```diff
@@ -38,10 +40,12 @@ public class LSPProgressManager {
 	private final Map<String, BlockingQueue<ProgressParams>> progressMap;

```

Comment:
Is it worth keeping all the params or only the last one? Is there value in processing intermediary reports when we have multiple ones to process.

Reply:
I thought about keeping all messages, but in the end if the progress is still running when the job starts, the job will pick some new report soon and then the work will be reflected. Thus for the time being I would to keep it simple. Maybe later someone  (or me I have time implements that)